tests/ui/assertions_on_constants.stderr

```diff
@@ -72,13 +72,5 @@ LL |     debug_assert!(true);
    |
    = help: remove it
 
-error: `assert!(true)` will be optimized out by the compiler
-  --> tests/ui/assertions_on_constants.rs:49:19
-   |
-LL |     const _: () = assert!(true);
```

Comment:
Yes, it is intended. The lint now doesn't fire on any const-assertions.

Reply:
We could catch the `true` literal value. As it sems like a programming typo?

Reply:
`assert!(true)` isn't much use, and is linted by `assertion_on_const` for that reason.

Looking at #12816, why do we want to skip linting `assert!(8 == (7+1));`? Is there a scenario where checking `8 == 7+1` makes sense?

Reply:
`8 == 7+1` is just an illustration for simple cases. It could be a more complex expression, like checking layout equality of different types. But I don't think checking if expressions is simple or complex is worth the complexity.
Also with the case of `8 == 7+1` get a typo, we have a compile-time panicking. So asserting `8 == 7+1` in const context is kinda worth it.

So maybe I could add a comment about this in test file?

Reply:
I think the layout case is handled by https://github.com/rust-lang/rust-clippy/pull/11966. It just doesn't cover correctly when it's in a block. Can we build on that and just add the block case to the check?

#11966 is the one that added this test case.

Reply:
It's done by 7b06d72